how-to/customize-workspace/client/src/share.ts

```diff
@@ -213,34 +212,36 @@ async function saveSharedWorkspace(workspaceId?:string) {
 
 async function saveShareRequest(payload) {
     try {
-        let uuid = window.crypto.randomUUID();
+        
+        const expiryInHours = 24;
+        let response = await requestResponse<any, { url: string, id?:string }>("share-save", payload);   
+        let id = response.id;
+        if(id === undefined) {
+            let indexOfId = response.url.lastIndexOf("/");
+            if(indexOfId !== -1) {                
+                id = response.url.substring(indexOfId + 1);
+            }
+        }
+
+        if(id === undefined) {
+            await notifyOfFailure("The share request you raised could not be generated.");
+            return;
+        }
+
         let platform = getCurrentSync();
         let platformInfo = await platform.Application.getInfo();
         let finsLink;
         
         if(platformInfo.manifestUrl.indexOf("https") === 0) {
-            finsLink = platformInfo.manifestUrl.replace("https", "fins") + "?$$shareId=" + uuid;
+            finsLink = platformInfo.manifestUrl.replace("https", "fins") + "?$$shareId=" + id;
         } else if(platformInfo.manifestUrl.indexOf("http") === 0){
-            finsLink = platformInfo.manifestUrl.replace("http", "fin") + "?$$shareId=" + uuid;
+            finsLink = platformInfo.manifestUrl.replace("http", "fin") + "?$$shareId=" + id;
```

Comment:
If you remove the `if` and just keep the `else` it will still work, `replace('http', 'fin')` would work for `https` as well, it would keep the `s` regardless

Reply:
updated